Make skip_exit_code configurable in BashOperator

jedcunningham opened the pull request:
Exit code 127 is used when a command is not found and we don't want to
skip those tasks. Exit code 100 was arbitrarily chosen, however, most
importantly it isn't used as a standard exit code:

https://tldp.org/LDP/abs/html/exitcodes.html

jedcunningham commented on airflow/operators/bash.py:
```diff
@@ -24,11 +24,13 @@
 from airflow.utils.decorators import apply_defaults
 from airflow.utils.operator_helpers import context_to_airflow_vars
 
+EXIT_CODE_SKIP = 100
```
I moved this back into the operator as it really is operator specific.

dstandish reviewed (approved):
No concerns from me, especially since this has not been released yet.

jedcunningham committed:
Change BashOperator skip exit code to 99

Exit code 127 is used when a command is not found and we don't want to
skip those tasks. Exit code 99 was arbitrarily chosen, however, most
importantly it isn't used as a standard exit code:

https://tldp.org/LDP/abs/html/exitcodes.html

This also allows users to provide their own `skip_exit_code` if they
want to use a different exit code than the default 99.

dstandish commented on tests/operators/test_bash.py:
```diff
@@ -117,7 +117,31 @@ def test_default_retries(self):
+        ):
+            BashOperator(
+                task_id='abc4', bash_command='set -e; echo "hello world"; exit 99;', skip_exit_code=None
+            ).execute({})
```
```suggestion
    @parameterized.expand(
        [
            (None, 99, AirflowSkipException),
            ({'skip_exit_code': 100}, 100, AirflowSkipException),
            ({'skip_exit_code': 100}, 101, AirflowException),
            ({'skip_exit_code': None}, 99, AirflowException),
        ]
    )
    def test_skip(self, extra_kwargs, actual_exit_code, expected_exc):
        match = (
            f"Bash command returned exit code {actual_exit_code}\\. Skipping\\."
            if expected_exc == AirflowSkipException
            else "Bash command failed\\. The command returned a non-zero exit code\\."
        )
        kwargs = dict(task_id='abc', bash_command=f'set -e; echo "hello world"; exit {actual_exit_code};')
        if extra_kwargs:
            kwargs.update(**extra_kwargs)
        with pytest.raises(expected_exc, match=match):
            BashOperator(**kwargs).execute({})
```

paramaterized here.... i think it makes it a lot easier to understand what's being tested

i think we could probably skip the `match` part cus it does add quite a bit of noise and i think the main thing we're verifying here is what exception gets raised but i left it in

dstandish replied:
i couldn't make the suggestion across the deleted line so if you accept this you'll need to remove the extraneous `def test_skip(self):`

jedcunningham replied:
Good call, thanks, and I whacked the `match` part too.

dstandish reviewed (approved):
i've made one suggestion for parameterizing test but looks good to me otherwise :)

jedcunningham committed:
Update tests/operators/test_bash.py

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>
